### ISSUE TITLE
Load and save text embeds

### ISSUE THREAD

**Issue by Metal-dude** [kijai/ComfyUI-WanVideoWrapper#1794]
Hello,

Congrats for your hard work (and the new WanVideo Sampler v2 ).

It would be a good thing to enable saving and loading text embeds to  a specific location of the drive.

It would enable to create and save a text embed once and for all, and just choose the location to load it once it is needed. It could be complementary or replace the current system of cache.

Another advantage is the possibility to create text embeds in separate workflows, so the text encoder model is not downloaded during the main sampling workflow.

Here is a piece of code to do that (il works fine provided a node is added after WanVideo TextEncodeSingleSaveToPath, but you know your code better and you could improve it).

    "WanVideoLoadTextEmbedsFromPath": WanVideoLoadTextEmbedsFromPath,
    "WanVideoTextEncodeSingleSaveToPath": WanVideoTextEncodeSingleSaveToPath,

    "WanVideoLoadTextEmbedsFromPath": "WanVideo LoadTextEmbedsFromPath",
    "WanVideoTextEncodeSingleSaveToPath": "WanVideo TextEncodeSingleSaveToPath",

class WanVideoLoadTextEmbedsFromPath:

    @classmethod
    def INPUT_TYPES(cls):
        return {"required": {
                "file_path": ("STRING", {"default": "C:/embeds.pt", "multiline": False})
            },
        }

    RETURN_TYPES = ("WANVIDEOTEXTEMBEDS", )
    RETURN_NAMES = ("text_embeds",)
    FUNCTION = "process"
    CATEGORY = "WanVideoWrapper"
    DESCRIPTION = "Loads a text embed file from a specific location of the computer."

    def process(self, file_path):

        if not os.path.exists(file_path):
            print(f"[Warning] The path `{file_path}` specified cannot be found.")
            return ()
        encoded_path = folder_paths.get_annotated_filepath(file_path)
        encoded = torch.load(encoded_path)
        prompt_embeds_dict = {
            "prompt_embeds": encoded,
            "negative_prompt_embeds": None,
            "echoshot": False
        }
        return (prompt_embeds_dict,)
    

class WanVideoTextEncodeSingleSaveToPath:

    @classmethod
    def INPUT_TYPES(s):
        return {"required": {
            "prompt": ("STRING", {"default": "", "multiline": True} ),
            "embeds_name_with_extension": ("STRING", {"default": 'embeds.pt', "multiline": False}),
            },
            "optional": {
                "t5": ("WANTEXTENCODER",),
                "force_offload": ("BOOLEAN", {"default": True}),
                "model_to_offload": ("WANVIDEOMODEL", {"tooltip": "Model to move to offload_device before encoding"}),
                "use_disk_cache": ("BOOLEAN", {"default": True, "tooltip": "Cache the text embeddings to disk for faster re-use, under the custom_nodes/ComfyUI-WanVideoWrapper/text_embed_cache directory"}),
                "device": (["gpu", "cpu"], {"default": "gpu", "tooltip": "Device to run the text encoding on."}),
            }
        }

    RETURN_TYPES = ("WANVIDEOTEXTEMBEDS", )
    RETURN_NAMES = ("text_embeds",)
    FUNCTION = "process"
    CATEGORY = "WanVideoWrapper"
    DESCRIPTION = "Encodes text prompt into text embedding, with the option to save it on a specific path of the output directory."

    def process(self, prompt, embeds_name_with_extension, t5=None, force_offload=True, model_to_offload=None, use_disk_cache=False, device="gpu"):
        # Unified cache logic: use a single cache file per unique prompt
        encoded = None
        echoshot = True if "[1]" in prompt else False
        if use_disk_cache:
            cache_dir = os.path.join(script_directory, 'text_embed_cache')
            os.makedirs(cache_dir, exist_ok=True)
            def get_cache_path(prompt):
                cache_key = prompt.strip()
                cache_hash = hashlib.sha256(cache_key.encode('utf-8')).hexdigest()
                return os.path.join(cache_dir, f"{cache_hash}.pt")
            cache_path = get_cache_path(prompt)
            if os.path.exists(cache_path):
                try:
                    log.info(f"Loading prompt embeds from cache: {cache_path}")
                    encoded = torch.load(cache_path)
                except Exception as e:
                    log.warning(f"Failed to load cache: {e}, will re-encode.")

        if t5 is None and encoded is None:
            raise ValueError("No cached text embeds found for prompts, please provide a T5 encoder.")

        if encoded is None:
            try:
                if model_to_offload is not None and device == "gpu":
                    log.info(f"Moving video model to {offload_device}")
                    model_to_offload.model.to(offload_device)
                    mm.soft_empty_cache()
            except:
                pass

            encoder = t5["model"]
            dtype = t5["dtype"]

            if device == "gpu":
                device_to = mm.get_torch_device()
            else:
                device_to = torch.device("cpu")

            if encoder.quantization == "fp8_e4m3fn":
                cast_dtype = torch.float8_e4m3fn
            else:
                cast_dtype = encoder.dtype
            params_to_keep = {'norm', 'pos_embedding', 'token_embedding'}
            for name, param in encoder.model.named_parameters():
                dtype_to_use = dtype if any(keyword in name for keyword in params_to_keep) else cast_dtype
                value = encoder.state_dict[name] if hasattr(encoder, 'state_dict') else encoder.model.state_dict()[name]
                set_module_tensor_to_device(encoder.model, name, device=device_to, dtype=dtype_to_use, value=value)
            if hasattr(encoder, 'state_dict'):
                del encoder.state_dict
                mm.soft_empty_cache()
                gc.collect()
            with torch.autocast(device_type=mm.get_autocast_device(device_to), dtype=encoder.dtype, enabled=encoder.quantization != 'disabled'):
                encoded = encoder([prompt], device_to)

            if force_offload:
                encoder.model.to(offload_device)
                mm.soft_empty_cache()

            # Save to cache if enabled
            if use_disk_cache:
                try:
                    if not os.path.exists(cache_path):
                        output_path = os.path.join(folder_paths.output_directory)
                        embeds_path = os.path.join(output_path, embeds_name_with_extension)
                        torch.save(encoded, embeds_path)
                        log.info(f"Saved prompt embeds to cache: {embeds_path}")
                except Exception as e:
                    log.warning(f"Failed to save cache: {e}")

        prompt_embeds_dict = {
            "prompt_embeds": encoded,
            "negative_prompt_embeds": None,
            "echoshot": echoshot
        }
        return (prompt_embeds_dict,)
